react/react1/week3/time-registration/src/SubmitShift.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import "./SubmitShift.css"
+import { useState } from 'react';
+
+const SubmitShift = ({addedShifts}) => {
+    const [values, setValues] = useState({
+        userName : "",
+        startDate : "",
+        endDate : ""
+    })
+    const [submited, setSubmit] = useState(false);
+      
+    const handelUserNameChange = (event) => {
+        setValues({...values, userName: event.target.value})
+    }
+    const handelStartDateChange = (event) => {
+        setValues({...values, startDate: event.target.value})
+    }
+    const handelEndDateChange = (event) => {
+        setValues({...values, endDate: event.target.value})
+    }
+    const handelSubmit = (event) => {
+        event.preventDefault();
+        setSubmit(true);
+        addedShifts(values.userName, values.startDate, values.endDate)
+      
+  
+    }
+ 
+    return (
+        <div className='formInput'>
+            <h1>Time Registration </h1>
+            <form className='registration-form'onSubmit={handelSubmit} >
+                <label>User Name :</label>
+                <input type="text" placeholder='User Name' name='userName' value={values.userName} onChange= {handelUserNameChange}/>
+                {submited && !values.userName ?  <span>Please enter your Name</span> : null }
```

Comment:
Here you can do this to avoid returning null:
`{submited && !values.userName && <span>Please enter your Name</span> }`

Reply:
Nice with the validation :)

Reply:
We could create a reusable validator here also:

```
const namesValidator = (input) => {

    const valdators = [input !== "",  input.length < 50]

    for (const validator of valdators) {
        if (!validator) return false;
    }
    return true;
}

function InputValidator({submited, value, validator}) {

    const [isValid, setValidation] = useState(false)

    useEffect(() => {
        setValidation(valid => validator(value) ? true : false)
    }, [value, validator])
   
    return (
        <>
           {submited && !isValid && <span>Please enter valid Name</span> }
        </> 
    )
}

```

And then use it like this:

```
  <label>User Name :</label>
  <input type="text" placeholder='User Name' name='userName' value={values.userName} onChange={handelUserNameChange}/>
  <InputValidator submited={submited} value={values.userName} validator={namesValidator}></InputValidator>
```
Replacing this:
  `  {/* {submited && !values.userName && <span>Please enter your Name</span> } */}`

Reply:
...although you also need a way to block the sbmit of the validation fails on any of the fields ... 🤔